app-config.yaml

```diff
@@ -122,6 +122,13 @@ kafka:
         - localhost:9092
 
 integrations:
+  gcs:
+    - host: 'storage.cloud.google.com'
```

Comment:
In the logic in the reader further down there's this code,

```ts
if (host !== 'storage.cloud.google.com') {
    throw new Error(`not a valid GCS URL: ${url}`);
  }
```

Therefore, it would seem that no matter what you do, the host MUST always be storage.cloud.google.com.  But perhaps the intent here is not that the HOST is customizable, but the path or URI on the end of a host is?  Given that, perhaps the name of this configuration setting might drive confusion.  I don't know what GCP calls it, but perhaps it should instead be like baseUrl or something that infers it is MORE than just a host (which is `storage` in this example, with `cloud.google.com` being the domain).

I also then see later the host is processed outside of the reader instantiation, so there may be a nuance I don't understand but didn't quite see why or how one method would check it but outside it would not?  (and to be fair, I don't understand how GCP works, just looking at the logic here :) )

Reply:
It's also hardcoded again on line 110 of the GscUrlReader seems.

Reply:
Thanks both, yes, what @freben mentioned is what I was trying to do: align with the other urlReaders. Let me take a look and see if I can make things more clear. I think the default service account might not work so I'll look into that.